utils/src/bitvec.rs

```diff
@@ -316,10 +316,13 @@ impl BitVec {
     #[inline(always)]
     fn mask_over_first_n_bits(num_bits: usize) -> Block {
         assert!(num_bits <= BITS_PER_BLOCK, "num_bits exceeds block size");
-        // Special-case for `BITS_PER_BLOCK` bits to avoid shl overflow
-        match num_bits {
-            BITS_PER_BLOCK => FULL_BLOCK,
-            _ => (1 << num_bits) - 1,
+        // Use a safer approach that avoids potential overflow
+        if num_bits == 0 {
```

Comment:
Since num_bits should never be 0, I think it would be better to simply document that precondition either in the function doc or through an assert_ne!(num_bits, 0).  (or both).